src/sentry/static/sentry/app/views/onboarding/onboarding.tsx

```diff
@@ -46,24 +53,25 @@ const ONBOARDING_STEPS = [
   },
 ];
 
-const stepShape = PropTypes.shape({
-  id: PropTypes.string,
-  title: PropTypes.string,
-  Component: PropTypes.func,
-});
-
-class Onboarding extends React.Component {
-  static propTypes = {
-    steps: PropTypes.arrayOf(stepShape),
-    projects: PropTypes.arrayOf(SentryTypes.Project),
-    organization: SentryTypes.Organization,
-  };
+type RouteParams = {
+  orgId: string;
+  step: string;
+};
 
+type Props = RouteComponentProps<RouteParams, {}> & {
+  steps: StepDescriptor[];
+  organization: Organization;
+  projects: Project[];
+};
+
+type State = StepData;
+
+class Onboarding extends React.Component<Props, State> {
   static defaultProps = {
```

Comment:
Should you also type this?
https://develop.sentry.dev/frontend/defaultprops/

Reply:
I think it's actually correctly typed. I think react inferred everything right here

Reply:
I am not getting any errors if I set default value of `steps` for example to `false`. Which is wrong, because steps should be of type `StepDescriptor[]` and not `boolean`. Also you can make a typo in `steps` without realizing it and it will not warn you that there is no such prop.

Reply:
oop, you're right, fixed!